cc/testing/evaluation/stochastictester/create_samples_mean.cc

```diff
@@ -0,0 +1,266 @@
+  for (int i=0; i<extra_values_length; i++) {
+    boundedmean->AddEntry(DiscretizeMean(extra_value,granularity));
+  }
+  return GetValue<double>(boundedmean->PartialResult().ValueOrDie());
+}
+
+// Creates a folder to contain all samples with a particular ratio value
+// (e.g., R95). Every folder contains 22 subfolders for each unique sample-pair.
+// Every subfolder contains seven runs of each sample-pair (14 files in total).
+void CreateSingleScenarioMean(int scenario, std::vector<double>valuesA,
+  std::vector<double>valuesB, double granularity, double epsilon,
+  int max_partitions, int max_contributions, int lower, int upper,
+  int number_of_samples, double ratio, double initial_value,
+  double extra_values_length, double extra_value) { 
+
+  double implemented_epsilon = epsilon / ratio;
+  std::string filepath = mean_samples_folder+"/R"
+    +std::to_string(static_cast<int>(ratio*100))+"/Scenario"
+    +std::to_string(scenario);
+  mkdir(filepath.c_str(), 0777);
+
+  for (int i=0; i<7; i++) {
+    std::ofstream samplefileA;
+    std::ofstream samplefileB;
+    samplefileA.open(filepath+"/TestCase"+std::to_string(i)+"A.txt");
+    samplefileB.open(filepath+"/TestCase"+std::to_string(i)+"B.txt");
+
+    if (extra_values_length == 0) {
+      for (int i=0; i<number_of_samples; i++) {
+        double outputA = BoundedMeanAlgorithm(valuesA, granularity,
+          implemented_epsilon, max_partitions, max_contributions, lower, upper);
+        samplefileA << outputA << "\n";
+        double outputB = BoundedMeanAlgorithm(valuesB, granularity,
+          implemented_epsilon, max_partitions, max_contributions, lower, upper);
+        samplefileB << outputB << "\n";
+      }
+      samplefileA.close();
+      samplefileB.close();
+    }
+    else {
+      for (int i=0; i<number_of_samples; i++) {
+      double outputA = BoundedMeanAlgorithm(valuesA, granularity,
+        implemented_epsilon, max_partitions, max_contributions, lower, upper);
+        samplefileA << outputA << "\n";
+      double outputB = LargeBoundedMeanAlgorithm(initial_value, extra_values_length,
+        extra_value, granularity, implemented_epsilon, max_partitions,
+        max_contributions, lower, upper);
+        samplefileB << outputB << "\n";
+      }
+      samplefileA.close();
+      samplefileB.close();
+    }
+  }   
+}
+// Runs each sample-pair with parameters that replicate those specified in:
+// https://github.com/google/differential-privacy/blob/main/proto/testing/bounded_mean_dp_test_cases.textproto.
+void GenerateAllScenariosMean(double ratio) {
+
+  const int num_of_samples = 100;
+  double small_epsilon = 0.1;
+  double default_epsilon = std::log(3);
+  double large_epsilon = 2*std::log(3);
+
+// Laplace noise, empty mean, default parameters
+  std::vector<double>zero_vector{0};
+  std::vector<double>valuesB1{1000};
+  CreateSingleScenarioMean(1,zero_vector,valuesB1,0.0078125,default_epsilon,1,1,0,1,num_of_samples,ratio);
```

Comment:
As mentioned elsewhere, we should re-think how we handle repeated calls like this. At the very least, perhaps storing the parameters in a single struct, and having an array of structs that we iterate over would be cleaner. Furthermore, these parameters should probably be input from a proto file (similar to how the Statistical Tester has them defined), so that the testing parameters can be modified without modifying the code. However, there may be better possible designs to consider.